awx/ui_next/src/components/FieldWithPrompt/FieldWithPrompt.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { bool, func, node, string } from 'prop-types';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { Tooltip } from '@patternfly/react-core';
+import { CheckboxField } from '@components/FormField';
+import { QuestionCircleIcon as PFQuestionCircleIcon } from '@patternfly/react-icons';
+import styled from 'styled-components';
+
+const QuestionCircleIcon = styled(PFQuestionCircleIcon)`
+  margin-left: 10px;
+`;
+
+const FieldHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding-bottom: var(--pf-c-form__label--PaddingBottom);
+
+  label {
+    --pf-c-form__label--PaddingBottom: 0px;
+  }
+`;
+
+const StyledCheckboxField = styled(CheckboxField)`
+  --pf-c-check__label--FontSize: var(--pf-c-form__label--FontSize);
+`;
+
+function FieldWithPrompt({
+  children,
+  fieldId,
+  i18n,
+  isRequired,
+  label,
+  promptId,
+  promptName,
+  promptValidate,
+  tooltip,
+  tooltipMaxWidth,
+}) {
+  return (
+    <div className="pf-c-form__group">
+      <FieldHeader>
+        <div>
+          <label className="pf-c-form__label" htmlFor={fieldId}>
+            <span className="pf-c-form__label-text">{label}</span>
+            {isRequired && (
+              <span className="pf-c-form__label-required" aria-hidden="true">
+                *
+              </span>
+            )}
+          </label>
+          {tooltip && (
+            <Tooltip
+              content={tooltip}
+              maxWidth={tooltipMaxWidth}
+              position="right"
+            >
+              <QuestionCircleIcon />
+            </Tooltip>
+          )}
+        </div>
+        <StyledCheckboxField
+          id={promptId}
+          label={i18n._(t`Prompt On Launch`)}
+          name={promptName}
+          validate={promptValidate}
+        />
```

Comment:
How does the value get tied into Formik? I can see that it's posting correctly, but I don't see how...?

Reply:
Admittedly my understanding of Formik is fairly shallow but it seems like the `name` prop is the key here.  When we initialize the form values we include `ask_job_type_on_launch` and then use that same key as the name of the `<Field>` surrounding the checkbox.

<img width="1679" alt="Screen Shot 2020-02-14 at 10 33 05 AM" src="https://user-images.githubusercontent.com/9889020/74544620-7095e280-4f15-11ea-9b18-bf2c6ee4394c.png">

There are two separate `<Field>` components that get rendered within the `FieldWithPrompt` component.  One for the field (job type dropdown in this case) and one for the prompt (ask_job_type_on_launch in this case).

Does that help?